server/controllers/submission.js

```diff
@@ -59,15 +59,28 @@ exports.getSubmissionByContest = asyncHandler(async (req, res) => {
       model: 'User',
       select: 'username profilePic'
     }
-  }).exec(async (err, { submissions }) => {
+  }).exec(async (err, contest) => {
     if (err) {
       res.setHeader('Content-Type', 'application/json');
       res.end(JSON.stringify({
         message: 'Failure'
       }));
       res.sendStatus(500);
     } else {
-      res.status(200).send(submissions)
+      if (contest) {
+        const structured = contest.submissions.map(item => ({
+          _id: item._id,
+          images: item.images,
+          contest: item.contest,
+          active: item.active,
+          artistName: item.artistId.username,
+          artistPic: item.artistId.profilePic,
+          artistId: item.artistId._id,
+        }))
+        res.status(200).send(structured)
+      } else {
+        res.status(304).send([])
```

Comment:
What is the reason for using a 304 here?

Reply:
The 304 was meant to communicate that the request went through successfully, but there were no results found and nothing was modified. After reading more into the 304 code, I believe a 204 would be a better choice here. What do you think?

Reply:
Yes, the 204 makes a lot more sense here

Reply:
Gotcha this is done.